### PR TITLE
Update artifact name and compile on Ubuntu 24

### PR DESCRIPTION
**Pull request by irenebm**

### CODE REVIEW

File: .github/workflows/reusable-workflow.yml
Comment (Tempate) [inline .github/workflows/reusable-workflow.yml:1]: Please update the `PLATFORM_SUPPORT.md`.

Comment (Tempate) [inline .github/workflows/reusable-workflow.yml:1]: There was also a temporary, ugly workaround on `TopicQoS` for Ubuntu 20.04. Try to get rid of it and see if it works in Ubuntu 22 & 24.

**Review by Tempate — CHANGES_REQUESTED** [review]
Thanks for taking the time to update the artifacts. It may be a hustle but it needs to be done.

**Review by Tempate — APPROVED** [review]
LGTM

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**